development-setup-wizard.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python3
+
+"""
+Development setup wizard
+
+This script leads you through the steps required to get a working
+development version of camphoric up and running.  It should reflect what is
+in the doc/development.md file.  There are a few environment variables that
+will affect the running of this script:
+
+- CAMPHORIC_SETUP_WIZARD_RECREATE_ENV: if this is undef, it will prompt, if
+  it is truthy, it will recreate, if it equals 'no', it will not recreate.
+- CAMPHORIC_SETUP_WIZARD_RECREATE_SECRET_KEY: if this is undef, it will prompt, if
+  it is truthy, it will recreate, if it equals 'no', it will not recreate.
+"""
+
+
+import argparse
+import os
+import os.path
+import re
+import shutil
+import subprocess
+import sys
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=__doc__.strip(),
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    parser.add_argument(
+        '-p',
+        '--preserve',
+        help='run non-destructively, non-interactively',
+        action='store_true')
+    args = parser.parse_args()
+
+    check_prereqs(args)
+    create_env_files(args)
+    check_node_modules_folder(args)
+    docker_compose_up(args)
+    set_env_values(args)
+    create_django_superuser(args)
+    load_sample_data(args)
+    run_tests(args)
+    finish(args)
+
+
+def check_prereqs(args):
+    if not shutil.which('docker'):
+        die(
+            'required command `docker` not found!',
+            'please make sure that you have it installed and try again')
+
+    hooray('Found all requirements')
+
+
+def create_env_files(args):
+    if os.environ.get('CAMPHORIC_SETUP_WIZARD_RECREATE_ENV') == 'no':
+        return
+
+    if args.preserve:
+        return
+
+    dir_name = './env'
+    if (os.path.isdir(dir_name)
+            and 'CAMPHORIC_SETUP_WIZARD_RECREATE_ENV' not in os.environ):
+        answer = input(' '.join([
```

Comment:
Here and elsewhere this "join" construct seems to be used to fold long string constants across lines.  Instead, just put the two strings on adjacent lines with no operator (no comma) between them; they will be read as one long string:
input(f'it looks like "{dir_name}" already exists, '
           'do you want to blow it away and start over? [y/n] ')